internal/config/image/cuda_image.go

```diff
@@ -219,6 +228,9 @@ func (i CUDA) OnlyFullyQualifiedCDIDevices() bool {
 // VisibleDevicesFromEnvVar returns the set of visible devices requested through
 // the NVIDIA_VISIBLE_DEVICES environment variable.
 func (i CUDA) VisibleDevicesFromEnvVar() []string {
+	if i.IsSwarmResource() {
+		return i.DevicesFromEnvvars(i.swarmResourceEnvvars...).List()
+	}
```

Comment:
The current behaviour in `container_config` is sublty different. We return the devices for the first ENVVAR that is set in the list that is set. I don't think that's what `DevicesFromEnvvars` does.

Reply:
Could you comment on this? Looking at the latest changes this was not addressed?